modules/search-pipeline-common/src/main/java/org/opensearch/search/pipeline/common/RenameResponseProcessor.java

```diff
@@ -0,0 +1,151 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.search.pipeline.common;
+
+import org.opensearch.action.search.SearchRequest;
+import org.opensearch.action.search.SearchResponse;
+import org.opensearch.common.bytes.BytesReference;
+import org.opensearch.common.collect.Tuple;
+import org.opensearch.common.document.DocumentField;
+import org.opensearch.common.xcontent.XContentHelper;
+import org.opensearch.core.xcontent.MediaType;
+import org.opensearch.core.xcontent.XContentBuilder;
+import org.opensearch.ingest.ConfigurationUtils;
+import org.opensearch.search.SearchHit;
+import org.opensearch.search.pipeline.Processor;
+import org.opensearch.search.pipeline.SearchRequestProcessor;
+import org.opensearch.search.pipeline.SearchResponseProcessor;
+
+import java.util.Map;
+
+/**
+ * This is a {@link SearchRequestProcessor} that renames a field before returning the search response
+ */
+public class RenameResponseProcessor extends AbstractProcessor implements SearchResponseProcessor {
+
+    private final String oldField;
+    private final String newField;
+    private final boolean ignoreMissing;
+
+    /**
+     * Key to reference this processor type from a search pipeline.
+     */
+    public static final String TYPE = "rename";
```

Comment:
May be `rename_response_field` or rename_field`, otherwise unclear really what's going to be renamed

Reply:
Resolved in [dec00b7](https://github.com/opensearch-project/OpenSearch/pull/7377/commits/dec00b795511cb2251e3ed5bc50c3d3eb339e5ee)